src/Geometry/Rectangle.ts

```diff
@@ -0,0 +1,18 @@
+import { Shape } from "./Shape";
+
+export class Rectangle extends Shape {
+    constructor(private width: number, private height: number) {
+        super();
+    }    
+    // Overloading
+    calculateArea(): number;
```

Comment:
it's not overload anything. I don't see several methods with same name here.

Reply:
I have added implementations to these functions.